src/Compilers/CSharp/Portable/Errors/LazyMissingNonNullTypesContextDiagnosticInfo.cs

```diff
@@ -22,7 +22,8 @@ internal LazyMissingNonNullTypesContextDiagnosticInfo(CSharpCompilation compilat
 
         protected override DiagnosticInfo ResolveInfo()
         {
-            return _type.IsValueType ? null : Symbol.ReportNullableReferenceTypesIfNeeded(_compilation, _context);
+            return !_type.IsNull && (_type.IsValueType || _type.IsErrorType()) ? null : Symbol.ReportNullableReferenceTypesIfNeeded(_compilation, _context);
```

Comment:
>_type.IsNull [](start = 20, length = 12)

Since the `_type.IsNull` check doesn't need to be delayed, can we do it up-front, and avoid queuing a lazy diagnostic in such cases?

Related to this, I didn't understand why we create some `LazyMissingNonNullTypesContextDiagnosticInfo` instances with `type: default` (which guarantee that no diagnostic is produced).